app/controllers/ValidateCommodityCodeController.scala

```diff
@@ -38,6 +39,7 @@ class ValidateCommodityCodeController @Inject() (
 
   def changeCategory(recordId: String): Action[AnyContent] =
     (identify andThen profileAuth andThen getData andThen requireData).async { implicit request =>
+      implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
       commodityService.isCommodityCodeValid(recordId).map {
         case true  =>
           Redirect(controllers.categorisation.routes.CategorisationPreparationController.startCategorisation(recordId))
```

Comment:
There changes remove the warning for  the import controllers.BaseController  when running unit tests